networkx/algorithms/hierarchy_coordinates.py

```diff
@@ -0,0 +1,1220 @@
+    nx_graphs: list of networkX Graphs
+        list of unweighted graphs produced from applying thresholds to the original weighted network
+
+    Examples
+    --------
+    Graphing the resultant network is recommended, as otherwise this is difficult to visualize...
+
+    See Also
+    --------
+    weight_nodes_by_condensation, graph_entropy, treeness
+
+    >>> import numpy as np
+    >>> a = np.array([
+    ...     [0, 0.2, 0, 0, 0],
+    ...     [0, 0, 0, 0.7, 0],
+    ...     [0, 0.4, 0, 0, 0],
+    ...     [0, 0, 0.1, 0, 1.0],
+    ...     [0, 0, 0, 0, 0],
+    ... ])
+    >>> condensed_networks, base_binary_networks = nx.node_weighted_condense(a)
+    >>> for network in condensed_networks:
+    ...     print(f'{network}, total weight: {nx.get_node_attributes(network, "weight")}')
+    DiGraph with 3 nodes and 2 edges, total weight: {0: 1, 1: 3, 2: 1}
+    DiGraph with 5 nodes and 4 edges, total weight: {0: 1, 1: 1, 2: 1, 3: 1, 4: 1}
+    DiGraph with 4 nodes and 3 edges, total weight: {0: 1, 1: 1, 2: 1, 3: 1}
+    DiGraph with 4 nodes and 3 edges, total weight: {0: 1, 1: 1, 2: 1, 3: 1}
+    DiGraph with 3 nodes and 2 edges, total weight: {0: 1, 1: 1, 2: 1}
+    DiGraph with 3 nodes and 2 edges, total weight: {0: 1, 1: 1, 2: 1}
+    DiGraph with 2 nodes and 1 edges, total weight: {0: 1, 1: 1}
+    DiGraph with 2 nodes and 1 edges, total weight: {0: 1, 1: 1}
+
+    Notes
+    ------
+    WIP TODO: As multiple independent graphs may form from applying threshold cutoffs to a weighted graph,
+    only the largest is considered. This might be worth considering in re-evaluating the meaning of
+    weighted network hierarchy coordinate evaluations. (See pages 7, 8 of [1]_, supplementary material)
+
+    An threshold_distribution of None results in a linear distribution, otherwise
+     the exponential distribution is sampled from exp(x) \in (0, 1)
+
+    .. [1] "On the origins of hierarchy in complex networks."
+     Corominas-Murtra, Bernat, Joaquín Goñi, Ricard V. Solé, and Carlos Rodríguez-Caso,
+     Proceedings of the National Academy of Sciences 110, no. 33 (2013)
+    """
+    import numpy as np
+
+    # binary check
+    if np.array_equal(np.unique(A), [0, 1]):
+        num_thresholds = 1
+    else:
+        num_thresholds = num_thresholds
+
+    # Establishing Thresholds
+    if num_thresholds == 1 or np.isclose(np.max(A) - np.min(A), 0, 1e-2):
+        nx_graphs = [nx.from_numpy_matrix(A, create_using=nx.DiGraph)]
+    else:
+        if threshold_distribution is None:
+            try:
+                thresholds = list(
+                    np.round(
+                        np.arange(
+                            np.min(A),
+                            np.max(A),
+                            (np.max(A - np.min(A))) / num_thresholds,
+                        ),
+                        4,
+                    )
+                )  # linear distribution
+            except:
```

Comment:
What kind of exception might occur here? Can we provide the exception type, i.e. `except ValueError:`?

Reply:
This was rendered obsolete by the earlier `np.isclose(np.max(A) - np.min(A), 0, 1e-15):` check, so I removed it.